UndertaleModTool/Scripts/Community Scripts/TouchControlsEnabler.csx

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Linq;
+using UndertaleModLib.Util;
+
+EnsureDataLoaded();
+
+string display_name = Data?.GeneralInfo?.DisplayName?.Content.ToLower();
+
+if (display_name != "deltarune chapter 1 & 2" && display_name != "deltarune chapter 1&2" && display_name != "undertale")
+{
+    ScriptError("Error 0: Only compatible with Undertale & Deltarune");
+    return;
+}
+
+string dataPath = Path.Combine(Path.GetDirectoryName(ScriptPath), "TouchControls_data");
+
+int lastTextPage = Data.EmbeddedTextures.Count - 1;
+int lastTextPageItem = Data.TexturePageItems.Count - 1;
+
+Dictionary<string, UndertaleEmbeddedTexture> textures = new Dictionary<string, UndertaleEmbeddedTexture>();
+
+UndertaleEmbeddedTexture controlsTexturePage = new UndertaleEmbeddedTexture()
+{
+    Name = new UndertaleString("Texture " + Data.EmbeddedTextures.Count),
+};
+controlsTexturePage.TextureData.TextureBlob = File.ReadAllBytes(Path.Combine(dataPath, "controls.png"));
```

Comment:
FWIW, mobile generally really likes texture pages that have a size with power of 2. Havent examined the size of your images to figure out whether that's the case currently.
Having texture pages that don't have a power-of-2-size can potentially lead to graphical issues and/or performance issues.

Reply:
idk what that means, clarify more.

Reply:
Not sure what "that" specifically is, so if you end up having questions after, just poke further at me

A "texture page" is just a synonym of an embedded texture.
"Power of 2" is a number that can be can be calculated by having 2 times X (with x being a number). Aka 1 (2⁰), 2 (2¹) , 4 (2²), 8 (2³), [...], 256 (2⁸), 512 (2⁹), 1024 (2¹⁰) etc.

Reply:
Not to the sprite - to the texture pages. Aka `controls.png`, as that image is what you're using as your texture page / embedded texture. 
Controls.png is 189x163 - both sides are not power of 2
Black.png is 640x480 - both sides are not power of 2.

You can combine both images to one page (1024 x 512):
![grafik](https://github.com/krzys-h/UndertaleModTool/assets/38186597/44cb3abf-65c8-45cc-8008-3c5d926ad7fb)

Reply:
I'm starting to understand, but this image u sent is 829x480, not 1024x512

Reply:
whoops, my image editor crops off transparency if possible when copy pasting.

Reply:
@Miepee send it to me on discord, I sent u a friend req

Reply:
not sure if I got it, it should look like this?
![test](https://github.com/krzys-h/UndertaleModTool/assets/42213155/22a338de-2c9c-43a8-af18-7ea4f1b21aa2)

Reply:
Yup, like that!

Reply:
Alrightyyyy, will do tomorrow.

Reply:
Done.